src/Cardano/Address.hs

```diff
@@ -144,6 +152,44 @@ class PaymentAddress key
             -- ^ Staking key
         -> Address
 
+-- | A 'StakingKeyPointer' type representing location of staking certificate
+-- in the blockchain. This can be achieved unambiguously by specifying
+-- slot number, transaction index and the index of certificate list.
+-- Alternatively, staking key can be used, like it is the case for
+-- 'DelegationAddress'.
+--
+-- @since 2.0.0
+data StakingKeyPointer = StakingKeyPointer
+    { slotNum :: Word64
+      -- ^ Pointer to the slot
```

Comment:
What about having this one as a `Natural` too ?

Reply:
Also, I think the concept of `pointer` is a bit more general than just for stake key. So I'd suggest to rename this to simply `Pointer` or `ChainPointer` and, `certificateIndex` to `outputIndex`.

Reply:
done in https://github.com/input-output-hk/cardano-addresses/pull/31/commits/bec1b0678ea08813980e8d243a76b95be8d478bb